lib/ReportDataCollector.php

```diff
@@ -130,20 +139,36 @@ public function getReportJson($options = JSON_PRETTY_PRINT, $depth = 512) {
 	}
 
 	/**
+	 * @param array $report
+	 */
+	public function addEventListenerReportData(&$report) {
+		foreach ($this->appConfigData as $index) {
+			foreach ($index as $innerKey => $innerVal) {
+				$report[$innerKey] = $innerVal;
```

Comment:
This change would help us. No if/else, just update the information. And no more updating of existing data. So in the enterprise_key app I have updated as :
1) https://github.com/owncloud/enterprise_key/pull/60/files#diff-33862b3d00ca988fd2abf700b704c8e0R103
2) https://github.com/owncloud/enterprise_key/pull/60/files#diff-33862b3d00ca988fd2abf700b704c8e0R106